### PR TITLE
some tests, tweaks

### PR DESCRIPTION
**Pull request by mstrap**
- I've made my initially new tests more rigid by asserting actual paths
- For the sake of completeness I've added index- and workdir-iterator tests for pathlists, but also including trees
- Two tiny issues I came across when debugging

Feel free to amend to already existing commits where it makes sense.

### CODE REVIEW

File: tests/iterator/tree.c [hunk 1]
```diff
@@ -1008,8 +1013,11 @@ void test_iterator_tree__pathlist_with_directory(void)
 	i_opts.pathlist.strings = (char **)filelist.contents;
 	i_opts.pathlist.count = filelist.length;
 
+	const char *expected2[] = { "subdir/subdir2/README", "subdir/subdir2/new.txt" };
+	size_t expected_len2 = 2;
```
Comment (ethomson) [inline tests/iterator/tree.c:1017]: Please move declarations up to the top of the block so that old MSVC doesn't choke on it.  :(

**Comment by ethomson** [comment]
Thanks for this.  There's no need to act on that comment about MSVC, I'll pull this into my branch and fix that up.  But something to consider for the future (we still support MSVC pre-2013, which don't allow variables to be declared in the middle of a block.  Sigh.)

**Comment by mstrap** [comment]
@ethomson Thanks, I recall these Travis (or AppVeyor) failures now from my last patches :/ For the future, is there a way to run these checks myself _before_ sending a pull request? I was starting to set up Travis and AppVeyor services for my own account, but that looks cumbersome.

**Comment by ethomson** [comment]
It is pretty cumbersome...  You can use `-Werror=declaration-after-statement` if your compiler supports it. (I don't remember when it was added, but surprisingly recently, IIRC.)

FWIW, I don't have travis or appveyor for github.com/ethomson/libgit2.  I tried it once and it was terrible.  I just have a lot of VMs.  Heh.  :(

**Pull request merged**